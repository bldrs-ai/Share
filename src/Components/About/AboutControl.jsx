import React, {useState} from 'react'
import Box from '@mui/material/Box'
import Typography from '@mui/material/Typography'
import {getCookieBoolean, setCookieBoolean} from '../../privacy/Privacy'
import useStore from '../../store/useStore'
import Dialog from '../Dialog'
import {ControlButton} from '../Buttons'
import AboutDescription from './AboutDescription'
import PrivacyControl from './PrivacyControl'
import AboutIcon from '../../assets/icons/Information.svg'
import LogoB from '../../assets/LogoB.svg'
import {Helmet} from 'react-helmet-async'


/**
 * Button to toggle About panel on and off
 *
 * @return {React.Component}
 */
export default function AboutControl() {
  const isAboutDialogSuppressed = useStore((state) => state.isAboutDialogSuppressed)
  const [isDialogDisplayed, setIsDialogDisplayed] = useState(getCookieBoolean({
    component: 'about',
    name: 'isFirstTime',
    defaultValue: true,
  }))
  const setIsDialogDisplayedLocal = (value) => {
    setIsDialogDisplayed(value)
  }

  const setIsDialogDisplayedForDialog = () => {
    setIsDialogDisplayed(false)
    setCookieBoolean({component: 'about', name: 'isFirstTime', value: false})
  }

  return (
    <ControlButton
      title='About'
      isDialogDisplayed={isDialogDisplayed}
      setIsDialogDisplayed={setIsDialogDisplayedLocal}
      icon={
        <AboutIcon
          style={{
            width: '20px',
            height: '20px',
          }}
        />
      }
      dialog={
        <AboutDialog
          isDialogDisplayed={isAboutDialogSuppressed ? false : isDialogDisplayed}
          setIsDialogDisplayed={setIsDialogDisplayedForDialog}
        />
      }
    />
  )
}


/**
 * The AboutDialog component
 *
 * @param {boolean} isDialogDisplayed
 * @param {Function} setIsDialogDisplayed
 * @return {React.ReactElement} React component
 */
export function AboutDialog({isDialogDisplayed, setIsDialogDisplayed}) {
  return (
    <Dialog
      icon={
        <Box sx={{display: 'inline-flex', flexDirection: 'column', textAlign: 'center'}}>
          <LogoB style={{width: '60px', height: '60px'}}/>
          <Typography variant='h6' sx={{marginLeft: '-9px', marginTop: '6px'}}>bldrs.ai</Typography>
        </Box>
      }
      headerText={''}
      isDialogDisplayed={isDialogDisplayed}
      setIsDialogDisplayed={setIsDialogDisplayed}
      content={<AboutContent setIsDialogDisplayed={setIsDialogDisplayed}/>}
      actionTitle='OK'
      actionCb={() => setIsDialogDisplayed(false)}
    />
  )
}


/**
 * The content portion of the AboutDialog
 *
 * @return {React.ReactElement} React component
 */
function AboutContent({setIsDialogDisplayed}) {
  return (
    <Box sx={{'& a': {textDecoration: 'none'}}}>
      <Helmet>
        <title>About â€” BLDRS</title>
      </Helmet>
      <Typography variant='h2' gutterBottom={true}>build every thing together</Typography>
      <a href='https://github.com/bldrs-ai/Share' target='_new'>
        github.com/bldrs-ai/Share
      </a>
      <AboutDescription setIsDialogDisplayed={setIsDialogDisplayed}/>
      <PrivacyControl/>
    </Box>)
}
