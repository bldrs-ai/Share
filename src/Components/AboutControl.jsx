import React, {useState, useEffect} from 'react'
import Slider from '@mui/material/Slider'
import Typography from '@mui/material/Typography'
import {makeStyles} from '@mui/styles'
import Dialog from './Dialog'
import debug from '../utils/debug'
import * as Privacy from '../privacy/Privacy'
import {ControlButton} from './Buttons'
import AboutIcon from '../assets/2D_Icons/Wave_person.svg'
import LogoB from '../assets/LogoB.svg'
import ShareIcon from '../assets/2D_Icons/Share.svg'
import OpenIcon from '../assets/2D_Icons/Open.svg'
import GitHubIcon from '../assets/2D_Icons/GitHub.svg'


/**
 * Button to toggle About panel on and off
 * @param {string} installPrefix For use in static asset links.
 * @return {Object} The AboutControl react component.
 */
export default function AboutControl({installPrefix}) {
  const [isDialogDisplayed, setIsDialogDisplayed] =
        useState(Privacy.getCookieBoolean({
          component: 'about',
          name: 'isFirstTime',
          defaultValue: true}))
  return (
    <ControlButton
      title='About BLDRS'
      icon={<AboutIcon/>}
      isDialogDisplayed={isDialogDisplayed}
      setIsDialogDisplayed={setIsDialogDisplayed}
      placement='top'
      dialog={
        <AboutDialog
          isDialogDisplayed={isDialogDisplayed}
          setIsDialogDisplayed={() => {
            setIsDialogDisplayed(false)
            Privacy.setCookieBoolean({component: 'about', name: 'isFirstTime', value: false})
          }}
          installPrefix={installPrefix}/>
      }/>
  )
}


/**
 * The AboutDialog component
 * @param {boolean} isDialogDisplayed
 * @param {function} setIsDialogDisplayed
 * @param {string} installPrefix node
 * @return {Component} React component
 */
function AboutDialog({isDialogDisplayed, setIsDialogDisplayed, installPrefix}) {
  return (
    <Dialog
      icon={<LogoB/>}
      headerText='BLDRS'
      isDialogDisplayed={isDialogDisplayed}
      setIsDialogDisplayed={setIsDialogDisplayed}
      content={<AboutContent installPrefix={installPrefix}/>}/>)
}


/**
 * The content portion of the AboutDialog
 * @param {string} installPrefix node
 * @return {Object} React component
 */
function AboutContent({installPrefix}) {
  const classes = useStyles()
  const [privacySlider, setPrivacySlider] = useState(0)
  useEffect(()=>{
    if (Privacy.isPrivacySocialEnabled()) {
      setPrivacySlider(20)
    } else if (Privacy.isPrivacyUsageEnabled()) {
      setPrivacySlider(10)
    } else {
      setPrivacySlider(0)
    }
  }, [])
  const marks = [
    {value: 0, label: 'Functional', info: 'Theme, UI state, cookie preference'},
    {value: 10, label: 'Usage', info: 'Stats from your use of Bldrs'},
    {value: 20, label: 'Social', info: 'Google\'s guess of your location and demographic'},
  ]
  const setPrivacy = (event) => {
    debug().log('AboutContent#setPrivacy: ', event.target.value)
    switch (event.target.value) {
      case 0:
        Privacy.setUsageAndSocialEnabled(false, false)
        setPrivacySlider(0)
        break
      case 10:
        Privacy.setUsageAndSocialEnabled(true, false)
        setPrivacySlider(10)
        break
      case 20:
        Privacy.setUsageAndSocialEnabled(true, true)
        setPrivacySlider(20)
        break
    }
  }

  return (
    <div className={classes.content}>
      <Typography
        variant='h4'
        gutterBottom={false}>Build Every Thing Together</Typography>
      <Typography gutterBottom={false} >We are open source ðŸŒ±<br/>
        <a href='https://github.com/bldrs-ai/Share' target='_new'>
          github.com/bldrs-ai/Share
        </a>
      </Typography>
      <ul>
        <li><OpenIcon/> View local IFC models</li>
        <li><GitHubIcon/> Open IFC models from GitHub</li>
        <li><ShareIcon/> Share IFC models</li>
      </ul>
      <Typography variant='h5' color='info'>Highlighted Projects:</Typography>
      <div className = {classes.demoContainer}>
        <a href={`${installPrefix}/share/v/gh/Swiss-Property-AG/Portfolio/main/KNIK.ifc#c:-12.84,3.53,9.64,-5.33,2.61,1.71`}>
          <img alt="Tinyhouse" src={`${installPrefix}/Tinyhouse.png`} className={classes.demo}/>
        </a>
        {/* eslint-disable-next-line */}
        <a href={`${installPrefix}/share/v/gh/IFCjs/test-ifc-files/main/Schependomlaan/IFC%20Schependomlaan.ifc#c:-19.95,17.97,25.31,4.52,0.65,1.24`}>
          <img
            alt="Schependomlaan"
            src={`${installPrefix}/Schependomlaan.png`}
            className={classes.demo}/>
        </a>
      </div>
      <div className={classes.settings}>
        <Typography variant='h5' color='info'>Privacy</Typography>
        <Slider
          onChange={setPrivacy}
          marks={marks}
          value={privacySlider}
          step={10}
          min={0}
          max={20}
          sx={{width: '80%', textAlign: 'center'}}/>
      </div>
    </div>)
}


const useStyles = makeStyles({
  content: {
    'minHeight': '300px',
    '& .MuiTypography-body1': {
      padding: '1em 0',
    },
    '& .MuiTypography-body2': {
      padding: '1em 0',
      opacity: 0.5,
    },
    '& ul': {
      width: '100%',
      margin: '0px',
      marginTop: '-10px',
      marginBottom: '15px',
      padding: '0px',
      textAlign: 'left',
      // TODO(pablo): appears to be removed but not sure why.  Here to
      // make sure.
      listStyleType: 'none',
    },
    '& li': {
      display: 'flex',
      justifyContent: 'flex-start',
      alignItems: 'center',
      margin: '0.5em',
      padding: '0px',
      fontWeight: 200,
      fontSize: '0.9em',
      // TODO(pablo): appears to be removed but not sure why.  Here to
      // make sure.
      listStyleType: 'none',
    },
    '& li svg': {
      width: '25px',
      height: '25px',
      marginRight: '0.5em',
    },
    '& a': {
      color: 'grey',
      paddingLeft: '4px',
      paddingRight: '4px',
      paddingBottom: '2px',
    },
  },
  version: {
    '@media (max-width: 900px)': {
      display: 'none',
    },
  },
  demo: {
    'height': '100px',
    'textAlign': 'center',
    'marginTop': '10px',
    'borderRadius': '10px',
    'boxShadow': 'rgba(50, 50, 93, 0.25) 0px 6px 12px -2px, rgba(0, 0, 0, 0.3) 0px 3px 7px -3px',
    '@media (max-width: 900px)': {
      height: '60px',
    },
  },
  demoContainer: {
    'display': 'flex',
    'flexDirection': 'row',
    'justifyContent': 'space-between',
    'height': '50px',
    '@media (max-width: 900px)': {
      height: '0px',
      justifyContent: 'center',
    },
  },
  settings: {
    'display': 'flex',
    'flexDirection': 'column',
    'justifyContent': 'center',
    'alignItems': 'center',
    'margin': '5em 0 0 0',
    'textAlign': 'center',
    'paddingTop': '20px',
    'borderTop': '1px solid lightGrey',
    '@media (max-width: 900px)': {
      paddingTop: '10px',
    },
    '& .MuiSlider-thumb': {
      backgroundColor: 'green',
      width: '15px',
      height: '15px',
    },
    '& .MuiSlider-track': {
      color: 'lightGray',
    },
    '& .MuiSlider-rail': {
      color: 'lightGray',
    },
  },
  toggle: {
    'width': '50px',
    '& .MuiSwitch-switchBase.Mui-checked': {
      'color': 'green',
      '&:hover': {
        backgroundColor: 'green',
      },
    },
    '& .MuiSwitch-switchBase.Mui-checked + .MuiSwitch-track': {
      backgroundColor: 'gray',
    },
    '& .MuiSwitch-thumb': {
      backgroundColor: 'lightGrey',
    },
  },
})
