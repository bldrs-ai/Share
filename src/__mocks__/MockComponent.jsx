import useTheme from '../Theme'
import {ThemeProvider} from '@mui/material/styles'
import React, {createContext} from 'react'


/**
 * @param {Object} children React component(s)
 * @return {Object} React component
 */
export const MockComponent = ({children}) => {
  const {theme, colorMode} = useTheme()

  return (
    <ColorModeContext.Provider value={colorMode}>
      <ThemeProvider theme={theme}>
        {children}
      </ThemeProvider>
    </ColorModeContext.Provider>
  )
}

const ColorModeContext = createContext({
  toggleColorMode: () => {},
})
