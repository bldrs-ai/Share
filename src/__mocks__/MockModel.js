import {Vector3} from 'three'

/* eslint-disable no-magic-numbers */
const ifcModel = {
  geometry: {
    boundingBox: {
      min: new Vector3(2, 2, 2),
      max: new Vector3(10, 10, 10),
      getCenter: () => {
        return new Vector3(-33, 15, -5.613276958465576)
      },
    },
  },
}

export default ifcModel
