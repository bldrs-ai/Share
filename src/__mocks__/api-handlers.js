import {rest} from 'msw'
import {
  MOCK_COMMENTS,
  MOCK_ISSUES,
  MOCK_ORGANIZATIONS,
  MOCK_REPOSITORY,
  MOCK_FILES,
} from '../utils/GitHub'


const httpOk = 200
const httpCreated = 201
const httpAuthorizationRequired = 401
const httpNotFound = 404

export const handlers = [
  rest.get('https://api.github.com/repos/:org/:repo/issues', (req, res, ctx) => {
    const {org, repo} = req.params

    if (org !== 'pablo-mayrgundter' || repo !== 'Share') {
      return res(ctx.status(httpNotFound))
    }

    return res(
        ctx.status(httpOk),
        ctx.json(MOCK_ISSUES.data),
    )
  }),

  rest.get('https://api.github.com/repos/:org/:repo/issues/:issueNumber/comments', (req, res, ctx) => {
    const {org, repo, issueNumber} = req.params

    if (org !== 'pablo-mayrgundter' || repo !== 'Share' || !issueNumber) {
      return res(ctx.status(httpNotFound))
    }

    return res(
        ctx.status(httpOk),
        ctx.json(MOCK_COMMENTS.data),
    )
  }),

  rest.get('https://api.github.com/repos/:org/:repo/contents/:path', (req, res, ctx) => {
    const {org, repo, path} = req.params
    const ref = req.url.searchParams.get('ref')

    if (org !== 'bldrs-ai' || repo !== 'Share' || path !== 'README.md') {
      return res(
          ctx.status(httpNotFound),
          ctx.json({
            message: 'Not Found',
            documentation_url: 'https://docs.github.com/rest/reference/repos#get-repository-content',
          }),
      )
    }

    let downloadURL = 'https://raw.githubusercontent.com/bldrs-ai/Share/main/README.md'

    if (ref === 'main') {
      downloadURL += '?token=MAINBRANCHCONTENT'
    } else if (ref === 'a-new-branch') {
      downloadURL += '?token=TESTTOKENFORNEWBRANCH'
    }

    return res(
        ctx.status(httpOk),
        ctx.set({
          'content-type': 'application/json; charset=utf-8',
        }),
        ctx.json({
          name: 'README.md',
          path: 'README.md',
          sha: 'a5dd511780350dfbf2374196d8f069114a7d9205',
          size: 1359,
          url: 'https://api.github.com/repos/bldrs-ai/Share/contents/README.md?ref=main',
          html_url: 'https://github.com/bldrs-ai/Share/blob/main/README.md',
          git_url: 'https://api.github.com/repos/bldrs-ai/Share/git/blobs/a5dd511780350dfbf2374196d8f069114a7d9205',
          download_url: downloadURL,
          type: 'file',
          content: 'U2hhcmUgaXMgYSB3ZWItYmFzZWQgQklNICYgQ0FEIGludGVncmF0aW9uIGVu\n' +
          'dmlyb25tZW50IGZyb20gW2JsZHJzLmFpXShodHRwczovL2JsZHJzLmFpLyku\n' +
          'CgotICpPcGVuKiBhbnkgSUZDIG1vZGVsIG9uIGdpdGh1YiBieSBwYXN0aW5n\n' +
          'IGludG8gdGhlIHNlYXJjaGJhciwgb3IgdXBsb2FkaW5nIGZyb20geW91ciBs\n' +
          'b2NhbCBkZXNrdG9wLgotICpWaWV3KiB0aGUgbW9kZWwsICpuYXZpZ2F0ZSog\n' +
          'aXRzIHN0cnVjdHVyZSBhbmQgdXNlICpjdXQgcGxhbmVzKiB0byB2aWV3IGlu\n' +
          'c2lkZS4KLSAqU2VhcmNoKiB0aGUgbW9kZWwncyBlbGVtZW50cyBhbmQgcHJv\n' +
          'cGVydGllcy4KLSAqQ29sbGFib3JhdGUqIHdpdGggdGVhbW1hdGVzIGJ5IGNv\n' +
          'bW1lbnRpbmcgb24gbW9kZWwgcGFydHMgYW5kIHByb3BlcnRpZXMgKGluIGRl\n' +
          'dmVsb3BtZW50KS4KLSAqU2hhcmUqIHdpdGggdGVhbW1hdGVzLCB1c2luZyBw\n' +
          'ZXJtYWxpbmtzIHRvIG1vZGVsIHBhcnRzIHdpdGggZXhhY3QgY2FtZXJhIHZp\n' +
          'ZXdzLgotICpFeHRlbmQqIG91ciBwbGF0Zm9ybSB3aXRoIHlvdXIgQXBwcy4g\n' +
          'KGluIGRldmVsb3BtZW50KQoKIVtpbWFnZV0oaHR0cHM6Ly91c2VyLWltYWdl\n' +
          'cy5naXRodWJ1c2VyY29udGVudC5jb20vMjQ4MDg3OS8xNzM1NDg3ODUtYzYx\n' +
          'YWM5NzYtNzUxZS00YTFmLWJhMjgtMTUxNGI0NGQ1MzllLnBuZykKCiMgQ29u\n' +
          'dHJpYnV0aW5nClBsZWFzZSBqb2luIGluIGNyZWF0aW5nIEJsZHJzISAgQ29t\n' +
          'ZSBjaGF0IHdpdGggdXMgYXQgdGhlIFtCbGRycyBEaXNjb3JkXShodHRwczov\n' +
          'L2Rpc2NvcmQuZ2cvYXBXSGZEdGtKcykuCgpJZiB5b3UgaGF2ZSBpZGVhcyBv\n' +
          'ciBpc3N1ZXMsIHBsZWFzZSBmaWxlIHRoZW0gaW4gb3VyIEdpdEh1YiBbaXNz\n' +
          'dWVzXShodHRwczovL2dpdGh1Yi5jb20vYmxkcnMtYWkvU2hhcmUvaXNzdWVz\n' +
          'KSBwYWdlLCBvciBtYWlsIGluZm9AYmxkcnMuYWkuCgojIyBEb25hdGlvbnMg\n' +
          'CklmIHlvdSB1c2UgQmxkcnMgZm9yIHByaXZhdGUgaG9zdGluZywgcGxlYXNl\n' +
          'IGNvbnRyaWJ1dGUgdG8gdGhlIFtCbGRycyBPcGVuIENvbGxlY3RpdmUgcHJv\n' +
          'amVjdF0oaHR0cHM6Ly9vcGVuY29sbGVjdGl2ZS5jb20vYmxkcnMpLgoKIyMg\n' +
          'RGV2ZWxvcG1lbnQKQmxkcnMgaXMgb3BlbiBzb3VyY2UgYW5kIHdlJ2QgYXBw\n' +
          'cmVjaWF0ZSB5b3VyIGhlbHAuCi0gW1Byb2plY3RzXShodHRwczovL2dpdGh1\n' +
          'Yi5jb20vb3Jncy9ibGRycy1haS9wcm9qZWN0cz9xdWVyeT1pcyUzQW9wZW4m\n' +
          'dHlwZT1iZXRhKQotIFtEZXNpZ24gRG9jXShodHRwczovL2dpdGh1Yi5jb20v\n' +
          'YmxkcnMtYWkvU2hhcmUvd2lraS9EZXNpZ24pCi0gW0RldmVsb3BlciBHdWlk\n' +
          'ZV0oaHR0cHM6Ly9naXRodWIuY29tL2JsZHJzLWFpL1NoYXJlL3dpa2kvRGV2\n' +
          'Oi1HdWlkZSkK\n',
          encoding: 'base64',
          _links: {
            self: 'https://api.github.com/repos/bldrs-ai/Share/contents/README.md?ref=main',
            git: 'https://api.github.com/repos/bldrs-ai/Share/git/blobs/a5dd511780350dfbf2374196d8f069114a7d9205',
            html: 'https://github.com/bldrs-ai/Share/blob/main/README.md',
          },
        }),
    )
  }),

  rest.post('https://api.github.com/repos/:org/:repo/issues', (req, res, ctx) => {
    const {org, repo} = req.params

    if (org !== 'bldrs-ai' || repo !== 'Share') {
      return res(
          ctx.status(httpNotFound),
          ctx.json({
            message: 'Not Found',
          }),
      )
    }

    return res(
        ctx.status(httpCreated),
    )
  }),

  rest.post('https://api.github.com/repos/:org/:repo/issues/:issueNumber/comments', (req, res, ctx) => {
    const {org, repo, issueNumber} = req.params

    if (org !== 'bldrs-ai' || repo !== 'Share' || !issueNumber) {
      return res(ctx.status(httpNotFound))
    }

    return res(
        ctx.status(httpCreated),
    )
  }),

  rest.patch('https://api.github.com/repos/:org/:repo/issues/:issueNumber', (req, res, ctx) => {
    const {org, repo} = req.params
    if (org !== 'pablo-mayrgundter' || repo !== 'Share' ) {
      return res(
          ctx.status(httpNotFound),
          ctx.json({
            message: 'Not Found',
          }),
      )
    }

    return res(
        ctx.status(httpOk),
    )
  }),

  rest.delete('https://api.github.com/repos/:org/:repo/issues/comments/:commentId', (req, res, ctx) => {
    const {org, repo, commentId} = req.params

    if (org !== 'bldrs-ai' || repo !== 'Share' || !commentId) {
      return res(ctx.status(httpNotFound))
    }

    return res(
        ctx.status(httpOk),
    )
  }),

  rest.get('https://api.github.com/user/orgs', (req, res, ctx) => {
    const authHeader = req.headers.get('authorization')

    if (!authHeader) {
      return res(
          ctx.status(httpAuthorizationRequired),
          ctx.json({
            message: 'Requires authentication',
            documentation_url: 'https://docs.github.com/rest/reference/orgs#list-organizations-for-the-authenticated-user',
          }),
      )
    }

    return res(
        ctx.status(httpOk),
        ctx.json(MOCK_ORGANIZATIONS.data),
    )
  }),

  rest.get('https://api.github.com/orgs/bldrs-ai/repos', (req, res, ctx) => {
    return res(
        ctx.status(httpOk),
        ctx.json({
          data: [MOCK_REPOSITORY],
        }),
    )
  }),

  rest.get('https://api.github.com/repos/:owner/:repo/contents', (req, res, ctx) => {
    return res(
        ctx.status(httpOk),
        ctx.json({
          data: [MOCK_FILES],
        }),
    )
  }),
]
