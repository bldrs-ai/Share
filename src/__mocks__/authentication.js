import {useAuth0} from '@auth0/auth0-react'


jest.mock('@auth0/auth0-react')


const mockGitHubUser = {
  name: 'Unit Testing',
  nickname: 'testing',
  picture: 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAEAAAABACAAAAACPAi4CAAAAwUlEQVR42u3VLQvCUBTGcT/LX6vMjyAIliVhYLELmgSzRbALA0EUVix' +
      'GQUEmFkG2riiO813EF+5Y8040nSed54ZfOOGegnyZggIKKKCAAj8DpnBOW4WhSINM3P8D8eGRBQTPIbYGXglhnXMHCihgD8zhlDaHkS2whMiUpIhvC0QQmLKBlS0gVTwzt' +
      '3Fu1sAEBu9xTLqCzwFpgetvj/tZE7wkB3Dtms+nc5EcgEjYq5VLTr2/yzxaAHqZFFBAAQXy5g5KPEV7KOa7LAAAAABJRU5ErkJggg==',
  email: 'testing@example.com',
  email_verified: true,
  sub: 'github|1234567',
  updated_at: '2023-02-22T17:07:29.123Z',
}

export const mockedUseAuth0 = jest.mocked(useAuth0, true)

export const mockedUserLoggedIn = {
  error: null,
  user: mockGitHubUser,
  isAuthenticated: true,
  isLoading: false,
  getAccessTokenSilently: jest.fn(),
  getAccessTokenWithPopup: jest.fn(),
  getIdTokenClaims: jest.fn(),
  loginWithRedirect: jest.fn(),
  loginWithPopup: jest.fn(),
  logout: jest.fn(),
}

export const mockedUserLoggedOut = {
  error: null,
  user: null,
  isAuthenticated: false,
  isLoading: false,
  getAccessTokenSilently: jest.fn(),
  getAccessTokenWithPopup: jest.fn(),
  getIdTokenClaims: jest.fn(),
  loginWithRedirect: jest.fn(),
  loginWithPopup: jest.fn(),
  logout: jest.fn(),
}
