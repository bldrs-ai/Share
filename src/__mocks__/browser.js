import {setupWorker} from 'msw'
import {initHandlers} from './api-handlers'


// This configures a Service Worker with the given request handlers.
export const worker = setupWorker(...initHandlers())
