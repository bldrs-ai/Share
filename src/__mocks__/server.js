import {setupServer} from 'msw/node'
import {handlers} from './api-handlers'

// This configures a request mocking server with the given request handlers.
export const server = setupServer(...handlers)
