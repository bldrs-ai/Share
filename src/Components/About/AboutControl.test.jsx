import React from 'react'
import {render, fireEvent, waitFor} from '@testing-library/react'
import {MockComponent} from '../../__mocks__/MockComponent'
import AboutControl from './AboutControl'


describe('About control tests', () => {
  test('renders the AboutControl button', () => {
    const {getByTitle} = render(<MockComponent><AboutControl/></MockComponent>)
    const aboutControl = getByTitle('About')
    expect(aboutControl).toBeInTheDocument()
  })

  test('renders AboutDialog when control is pressed', () => {
    const {getByTitle, getByText} = render(<MockComponent><AboutControl/></MockComponent>)
    const aboutControl = getByTitle('About')
    fireEvent.click(aboutControl)
    const dialogTitle = getByText('build every thing together')
    expect(dialogTitle).toBeInTheDocument()
  })

  it('updates the title when the dialog is open', async () => {
    render(<AboutControl/>, {
      wrapper: MockComponent,
    })

    await(waitFor(() => expect(document.title).toBe('About â€” BLDRS')))
  })
})
