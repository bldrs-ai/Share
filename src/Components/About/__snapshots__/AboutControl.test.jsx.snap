// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutControl renders AboutDialog when control is pressed and updates the document title 1`] = `
<div>
  <button
    aria-pressed="false"
    class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeSmall MuiToggleButton-success css-1aw5b4q-MuiButtonBase-root-MuiToggleButton-root"
    data-mui-internal-clone-element="true"
    data-testid="control-button-about"
    tabindex="0"
    title="About Bldrs
1.0.1661"
    type="button"
    value=""
    variant="control"
  >
    <div
      class="MuiBox-root css-1dukxs6"
    >
      <svg
        data-filename="src/assets/LogoB.svg"
      />
    </div>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
